### PR TITLE
Fe analysis update

### PR DESCRIPTION
**Pull request by kgrunthal**
1. Adapted code to use the '.solve' function of the `N`-matrix objects. this allows using it with PTA models containing `ECORR` noise

2. Former code calculated and subtracted the object `XNTSigmaTNy` twice, leading to too small values of the Fe-statistic. This mistake is solved. Now the code matches the calculations outlined in Ellis, Siemens, Creighton 2012.

Note: The updated version is an EPTA contribution to IPTA DR3. It is used in EPTA and MPTA data analyses.

### CODE REVIEW

File: enterprise_extensions/frequentist/Fe_statistic.py [hunk 1]
```diff
@@ -15,49 +30,23 @@ class FeStat(object):
 
     """
 
-    def __init__(self, psrs, params=None):
+    def __init__(self, psrs, params=None, custom_models={}, inc_crn=False, orf=None, pta=None):
```
Comment (vhaasteren) [inline enterprise_extensions/frequentist/Fe_statistic.py:33]: It looks like the psrs aren't used anymore to build the pta object. That doesn't look like it's compatible with how things were. Is this really necessary? Can we not just allow both ways of creating the FeStat object? This is not backward compatible

File: enterprise_extensions/frequentist/Fe_statistic.py [hunk 2]
```diff
@@ -187,58 +197,67 @@ def compute_Fe(self, f0, gw_skyloc, brave=False, maximized_parameters=False):
...
-        :param TNx: T^T N^{-1} x precomputed
-        :param TNy: T^T N^{-1} y precomputed
+
         :return: inner product (x|y)
         """
 
     # white noise term
+    """
+    -- before: --
     Ni = Nmat
```
Comment (vhaasteren) [inline enterprise_extensions/frequentist/Fe_statistic.py:225]: Probably best not to have these kind of comments in committed code

File: enterprise_extensions/frequentist/Fe_statistic.py [hunk 3]
```diff
@@ -1,9 +1,24 @@
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
+"""
+Feb 2019, Bence Becsy (NANOGrav):   created file, implemented FeStatistics
+
+Dec 2023, Kathrin Grunthal (EPTA):  code not suitable for Sherman-Morrison
```
Comment (vhaasteren) [inline enterprise_extensions/frequentist/Fe_statistic.py:6]: Does it work with Sherman-Morrison now?

**Review by vhaasteren — CHANGES_REQUESTED** [review]
This looks mostly ok. However, based on the fact that now the pta object is passed rather than made, and that psrs is still required but no longer used the same way, I'd say this is not backward compatible.

Am I seeing that right?

File: enterprise_extensions/frequentist/Fe_statistic.py (hunk 1 above)
Comment (kgrunthal) [reply enterprise_extensions/frequentist/Fe_statistic.py:37]: Yes, totally possible. The new pull request has a more versatile if-statement

File: enterprise_extensions/frequentist/Fe_statistic.py (hunk 3 above)
Comment (kgrunthal) [reply enterprise_extensions/frequentist/Fe_statistic.py:6]: Yes!!!

**Pull request merged**